primitives/consensus/slots/src/lib.rs

```diff
@@ -124,6 +124,12 @@ impl SlotDuration {
 	pub const fn as_duration(&self) -> sp_std::time::Duration {
 		sp_std::time::Duration::from_millis(self.0)
 	}
+
+	/// Initialise from given [`sp_std::time::Duration`].
+	pub fn from_duration(duration: sp_std::time::Duration) -> Option<Self> {
+		let millis: u64 = duration.as_millis().try_into().ok()?;
+		Some(Self::from_millis(millis))
+	}
```

Comment:
Can be `From<Duration>`.

And `as_millis() as u64` is fine here. Until this overflows, there still seems to be a lot of time :P

Reply:
Someone likes to live dangerously :P I agree though!

Reply:
Whoever reads this in the future with Substrate still running. Sorry :P